### PR TITLE
Added url generate service

### PR DESCRIPTION
**Pull request by AlexeyShok**
Added service for URL generation. Using AES encryption for generate URL. Endpoints for hotel confirmation page now receive encrypted reference code. Secrets added to Vault.

### CODE REVIEW

File: Api/Services/PropertyOwners/UrlGenerationService.cs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            if (cipherText == null || cipherText.Length <= 0)
+                throw new ArgumentNullException("cipherText");
+            if (Key == null || Key.Length <= 0)
+                throw new ArgumentNullException("Key");
+            if (IV == null || IV.Length <= 0)
+                throw new ArgumentNullException("IV");
+
+            // Declare the string used to hold
+            // the decrypted text.
+            string plaintext = null;
```
Comment (kirillta) [inline Api/Services/PropertyOwners/UrlGenerationService.cs:94]: `plainText`

File: Api/Services/PropertyOwners/UrlGenerationService.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+        public string Generate(string referenceCode)
+        {
+            var variablePartOfUrl = Encrypt(referenceCode);
+
+            return $"{_urlGenerationOptions.ConfirmationPageUrl}/{variablePartOfUrl}";
+        }
+
+
+        public string Decrypt(string encryptedString)
```
Comment (kirillta) [inline Api/Services/PropertyOwners/UrlGenerationService.cs:25]: `Generate`/`Decrypt` isn't look like a pair actions. Besides we pass only a reference code to `Generate`. I suggest to rename either `Decrypt` to `ReadReferenceCode`, or `Generate` to `Encrypt`

**Review by kirillta — APPROVED** [review]

File: Api/Services/PropertyOwners/UrlGenerationService.cs (hunk 1 above)
Comment (AlexeyShok) [reply Api/Services/PropertyOwners/UrlGenerationService.cs]: Renamed

File: Api/Services/PropertyOwners/UrlGenerationService.cs (hunk 2 above)
Comment (AlexeyShok) [reply Api/Services/PropertyOwners/UrlGenerationService.cs]: Renamed to ReadReferenceCode

**Pull request merged**